src/app/betting/feature/place-bet-page.component.ts

```diff
@@ -671,4 +673,11 @@ export default class PlaceBetPageComponent {
       { allowSignalWrites: true },
     );
   }
+
+  ngOnInit() {
+    this.route.queryParamMap.subscribe((params) => {
+      const betTypeParam = params.get('betType') ?? '';
+      this.selectedBetType.set(getBetType(betTypeParam));
+    });
+  }
```

Comment:
Instead of subscribing to the route, we can use an input (this is new in Angular 16). Manually subscribing to Observables can cause memory leaks if not handled properly.

The input name needs to match the query parameter name. I think it would go something like this:

```typescript
private readonly betType = input<BetType>();
```

Then, you can access the value of `betType` in `ngOnInit`.